File: web/src/main/java/top/bettercode/summer/web/exception/DisabledUserException.java
package top.bettercode.summer.web.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

/**
 * 找不到资源
 *
 * @author Peter Wu
 */
@ResponseStatus(value = HttpStatus.UNAUTHORIZED, reason = "disabledUser")
public class DisabledUserException extends RuntimeException {

  private static final long serialVersionUID = 1L;

  public DisabledUserException() {
    this("disabledUser");
  }

  public DisabledUserException(String message) {
    this(message, null);
  }

  public DisabledUserException(String message, Throwable cause) {
    super(message, cause);
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/exception/DisabledUserException.kt
package top.bettercode.summer.web.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

/**
 * 找不到资源
 *
 * @author Peter Wu
 */
@ResponseStatus(value = HttpStatus.UNAUTHORIZED, reason = "disabledUser")
class DisabledUserException @JvmOverloads constructor(message: String? = "disabledUser", cause: Throwable? = null) : RuntimeException(message, cause) {
}